### PR TITLE
Add workaround for https://issuetracker.google.com/issues/63649622

### LINKED ISSUE #3438: Android O CastException
For Android O i noticed such crash in crashlytic,
<pre>Fatal Exception: java.lang.ClassCastException: android.system.UnixSocketAddress cannot be cast to java.net.InetSocketAddress
       at libcore.io.IoBridge.getLocalInetSocketAddress(IoBridge.java:700)
       at libcore.io.IoBridge.connectDetail(IoBridge.java:197)
       at libcore.io.IoBridge.connect(IoBridge.java:138)
       at java.net.PlainSocketImpl.socketConnect(PlainSocketImpl.java:129)
       at java.net.AbstractPlainSocketImpl.doConnect(AbstractPlainSocketImpl.java:356)
       at java.net.AbstractPlainSocketImpl.connectToAddress(AbstractPlainSocketImpl.java:200)
       at java.net.AbstractPlainSocketImpl.connect(AbstractPlainSocketImpl.java:182)
       at java.net.SocksSocketImpl.connect(SocksSocketImpl.java:356)
       at java.net.Socket.connect(Socket.java:616)
       at okhttp3.internal.platform.AndroidPlatform.connectSocket(AndroidPlatform.java:63)
       at okhttp3.internal.connection.RealConnection.connectSocket(RealConnection.java:223)
       at okhttp3.internal.connection.RealConnection.connect(RealConnection.java:149)
       at okhttp3.internal.connection.StreamAllocation.findConnection(StreamAllocation.java:192)
       at okhttp3.internal.connection.StreamAllocation.findHealthyConnection(StreamAllocation.java:121)
       at okhttp3.internal.connection.StreamAllocation.newStream(StreamAllocation.java:100)
       at okhttp3.internal.connection.ConnectInterceptor.intercept(ConnectInterceptor.java:42)
       at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
       at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
       at okhttp3.internal.cache.CacheInterceptor.intercept(CacheInterceptor.java:93)
       at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
       at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
       at okhttp3.internal.http.BridgeInterceptor.intercept(BridgeInterceptor.java:93)
       at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
       at okhttp3.internal.http.RetryAndFollowUpInterceptor.intercept(RetryAndFollowUpInterceptor.java:120)
       at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
       at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
       at com.pubnub.api.interceptors.SignatureInterceptor.intercept(SignatureInterceptor.java:32)
       at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:92)
       at okhttp3.internal.http.RealInterceptorChain.proceed(RealInterceptorChain.java:67)
       at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:185)
       at okhttp3.RealCall$AsyncCall.execute(RealCall.java:135)
       at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1162)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:636)
       at java.lang.Thread.run(Thread.java:764)
</pre>
Will this be fixed in new version of library?

### PR DESCRIPTION
**Pull request by TheMarketka**
This fixes #3438

### CODE REVIEW

File: okhttp/src/main/java/okhttp3/internal/platform/AndroidPlatform.java [hunk 1]
```diff
@@ -76,6 +77,13 @@
       IOException ioException = new IOException("Exception in connect");
       ioException.initCause(e);
       throw ioException;
+    } catch (ClassCastException e) {
+      // On android 8.0, socket.connect throws a ClassCastException due to a bug
+      if (Build.VERSION.SDK_INT == 26) {
+        log(WARN, "Swallowed ClassCastException, see https://issuetracker.google.com/issues/63649622", e);
```
Comment (yschimke) [inline okhttp/src/main/java/okhttp3/internal/platform/AndroidPlatform.java]: Without seeing this exact Android code, it seems hard to review.  But I don't think this is correct based on previous versions.  It seems like the CCE is thrown while processing a root IOException.  I'm basing this on connectDetail always going into an Exception.

At minimum swallowing only seems correct if the connect is successful (although later operations will fail). 

@JakeWharton do you know anyone who could check the exact state after this exception?

Comment (TheMarketka) [reply okhttp/src/main/java/okhttp3/internal/platform/AndroidPlatform.java]: @yschimke https://android.googlesource.com/platform/libcore/+/oreo-release/luni/src/main/java/libcore/io/IoBridge.java#700 there is the Android code.

Comment (TheMarketka) [reply okhttp/src/main/java/okhttp3/internal/platform/AndroidPlatform.java]: @yschimke and for reference, there is the fixed version of the Android code, which did not make it to the initial release: https://android.googlesource.com/platform/libcore/+/master/luni/src/main/java/libcore/io/IoBridge.java

Comment (TheMarketka) [reply okhttp/src/main/java/okhttp3/internal/platform/AndroidPlatform.java]: tl;dr: rethrowing the exception as IOException seems about right.

**Comment by swankjesse** [comment]
Nice. Thank you!

**Pull request merged**